packages/firestore/src/local/memory_query_cache.ts

```diff
@@ -163,7 +201,14 @@ export class MemoryQueryCache implements QueryCache {
     targetId: TargetId
   ): PersistencePromise<void> {
     this.references.addReferences(keys, targetId);
-    return PersistencePromise.resolve();
+    const referenceDelegate = this.persistence.referenceDelegate;
+    const promises: Array<PersistencePromise<void>> = [];
+    if (referenceDelegate) {
+      keys.forEach(key => {
+        promises.push(referenceDelegate.addReference(txn, key));
+      });
+    }
+    return PersistencePromise.waitFor(promises);
```

Comment:
If the set of keys here is small, you could also do:

```
return PersistencePromise.forEach(keys.toArray(), key -> referenceDelegate.addReference(txn, key));
```

Reply:
I don't know if it's small? It's however many keys are added to a target.